### PR TITLE
Work out user controller implementation

### PR DESCRIPTION
**Pull request by siemensikkema**
- split up code into dedicated files
- only accept validated input for user init/update
- hash passwords
- update dependencies
- reduce use of "string" literals
- add update route

### CODE REVIEW

**Comment by siemensikkema** [comment]
@steffendsommer it turns out, the BCryptHasher already includes salt. So if you use `drop.hash` for the `UserController` and configure it correctly it should be secure enough.

File: Sources/JWTKeychain/Validators/Validators.swift [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    public func validate(_ input: String) throws {
+        // TODO: decide on good standards for name validation. 
+        guard input.characters.count > 0, input.characters.count < 256 else {
+            throw error("Name should be between 1 and 255 characters in length")
+        }
+    }
+}
+
+// TODO: replace with Sugar's version when it's available
+public final class StrongPassword: Validator {
```
Comment (steffendsommer) [inline Sources/JWTKeychain/Validators/Validators.swift:17]: StrongPassword + Unique + Exists are in Sugar now and some of them might also be vapor/validation vapor/fluent-validation (I haven't checked for an overlap yet)

File: Sources/JWTKeychain/Models/Users/User.swift [hunk 2]
```diff
@@ -132,42 +158,61 @@ extension User: EmailAddressRepresentable {
...
+
+        return try self.init(
+            email: Valid(email),
+            name: name.map(Valid.init),
+            password: hasher.hash(Valid(password)))
     }
 
-    public static func update(request: Request) throws -> Self {
+    /// Updates an existing user with the values from the request. Hashes password using the hasher in case of a
+    /// password change .
```
Comment (steffendsommer) [inline Sources/JWTKeychain/Models/Users/User.swift:186]: There's an extra space after "change"

**Review by steffendsommer — APPROVED** [review]

File: Sources/JWTKeychain/Validators/Validators.swift (hunk 1 above)
Comment (siemensikkema) [reply Sources/JWTKeychain/Validators/Validators.swift:17]: let's do this in another PR as well

**Pull request merged**